corehq/apps/app_manager/README.md

```diff
@@ -1,21 +1,57 @@
-# Application terminology
+# App manager terminology
 
-For any given application, there are a number of different documents.
+## Applications (and builds)
 
-The primary application document is an instance of `Application`.  This
-document's id is what you'll see in the URL on most app manager pages. Primary
-application documents should have `copy_of == None` and `is_released ==
-False`. When an application is saved, the field `version` is incremented.
+An application typically has many different `Application` documents: 
+one for the current/primary/canonical application, plus one for each saved build.
+The current app's id is what you'll see in the URL on most app manager pages;
+most pages will redirect to the current app if given the id of an older build.
+
+In saved builds, `copy_of` contains the primary app's id, while it's `None` for the primary app itself.
+The property `master_id` gets the id of the primary app doc, whether it's called on a saved build or on the primary app.
```

Comment:
this is in reference to Linked Applications right? I think we should specify that

Reply:
This is not related to linked apps, though I agree it's confusing that they booth use "master" - this is why linked apps now use `upstream_app_id`.

Reply:
hmm, i am getting confused between `copy_of` and `master_id`. It's not clear why are these two different. "primary app's id" and "id of the primary app doc" appear to be the same.

Reply:
`_id` is for primary apps, `copy_of` is for builds, and `master_id` is for code that has to handle both primary apps and builds